domain/schema/model/sql/0007-charm.sql

```diff
@@ -1,27 +1,398 @@
+    PRIMARY KEY (charm_uuid, name)
+);
+
+CREATE INDEX idx_charm_extra_binding_charm
+ON charm_extra_binding (charm_uuid);
+
+CREATE TABLE charm_category (
+    charm_uuid          TEXT NOT NULL,
+    value               TEXT NOT NULL,
+    CONSTRAINT          fk_charm_category_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    PRIMARY KEY (charm_uuid, value)
 );
 
+CREATE INDEX idx_charm_category_charm
+ON charm_category (charm_uuid);
+
+CREATE TABLE charm_tag (
+    charm_uuid          TEXT NOT NULL,
+    value               TEXT NOT NULL,
+    CONSTRAINT          fk_charm_tag_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    PRIMARY KEY (charm_uuid, value)
+);
+
+CREATE INDEX idx_charm_tag_charm
+ON charm_tag (charm_uuid);
+
+CREATE TABLE charm_storage_kind (
+    id       INT PRIMARY KEY,
+    name     TEXT NOT NULL
+);
+
+INSERT INTO charm_storage_kind VALUES
+    (0, 'block'),
+    (1, 'filesystem');
+
 CREATE TABLE charm_storage (
-    charm_uuid       TEXT NOT NULL,
-    name             TEXT NOT NULL,
-    description      TEXT,
-    storage_kind_id  INT NOT NULL,
-    shared           BOOLEAN,
-    read_only        BOOLEAN,
-    count_min        INT NOT NULL,
-    count_max        INT NOT NULL,
-    minimum_size_mib INT,
-    location         TEXT,
-    CONSTRAINT       fk_storage_instance_kind
-        FOREIGN KEY  (storage_kind_id)
-        REFERENCES   storage_kind(id),
-    CONSTRAINT       fk_charm_storage_charm
-        FOREIGN KEY  (charm_uuid)
-        REFERENCES   charm(uuid),
+    charm_uuid          TEXT NOT NULL,
+    name                TEXT NOT NULL,
+    description         TEXT,
+    storage_kind_id     INT NOT NULL,
+    shared              BOOLEAN,
+    read_only           BOOLEAN,
+    count_min           INT NOT NULL,
+    count_max           INT NOT NULL,
+    minimum_size_mib    INT,
+    location            TEXT,
+    CONSTRAINT          fk_charm_storage_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    CONSTRAINT          fk_charm_storage_charm_storage_kind
+        FOREIGN KEY     (storage_kind_id)
+        REFERENCES      charm_storage_kind(id),
     PRIMARY KEY (charm_uuid, name)
 );
 
 CREATE INDEX idx_charm_storage_charm
 ON charm_storage (charm_uuid);
+
+CREATE TABLE charm_storage_property (
+    charm_uuid          TEXT NOT NULL,
+    charm_storage_name  TEXT NOT NULL,
+    key                 TEXT,
+    value               TEXT,
+    CONSTRAINT          fk_charm_storage_property_charm
+        FOREIGN KEY     (charm_uuid)
+        REFERENCES      charm(uuid),
+    CONSTRAINT          fk_charm_storage_property_charm_storage
+        FOREIGN KEY     (charm_storage_name)
+        REFERENCES      charm_storage(name),
+    PRIMARY KEY (charm_uuid, charm_storage_name, key)
+);
+
+CREATE INDEX idx_charm_storage_property_charm
+ON charm_storage_property (charm_uuid);
+
+CREATE TABLE charm_device (
```

Comment:
This was ostensibly for k8s charm and I am not sure if it's used. We should audit what we have on ch and see if we still need this.

Reply:
I'm leaving for now after our discussion.